src/mon/MDSMonitor.cc

```diff
@@ -860,6 +839,32 @@ bool MDSMonitor::prepare_beacon(MonOpRequestRef op)
     }
   }));
 
+  return true;
+
+evict:
+  if (!mon.osdmon()->is_writeable()) {
+    dout(1) << __func__ << ": waiting for writeable OSDMap to evict" << dendl;
+    mon.osdmon()->wait_for_writeable(op, new C_RetryMessage(this, op));
+    return false;
+  }
+
+  fail_mds_gid(pending, gid);
```

Comment:
`fail_mds_gid` evicts (blocklists) if the mds is currently not in `standby-replay` state. Seems like we can reach here in that case.

Reply:
Ya, I think the choice was made not to blocklist standby/standby-replay daemons. Do you suggest a change?

Reply:
> Ya, I think the choice was made not to blocklist standby/standby-replay daemons. Do you suggest a change?

Not sure why the choice was made, but if the standby(-replay) mds is misbehaving then blocklisting it seems to be the right thing to do.